packages/vincent-contracts/src/LibVincentDiamondStorage.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "./diamond-base/libraries/LibDiamond.sol";
+import "./IPKPNftFacet.sol";
+
+library VincentAppStorage {
+    using EnumerableSet for EnumerableSet.UintSet;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    bytes32 internal constant APP_STORAGE_SLOT = keccak256("lit.vincent.app.storage");
+
+    struct VersionedApp {
+        EnumerableSet.Bytes32Set toolIpfsCidHashes;
+        EnumerableSet.UintSet delegatedAgentPkps;
+        uint256 version;
+        bool enabled;
+    }
+
+    struct App {
+        EnumerableSet.AddressSet delegatees;
+        EnumerableSet.Bytes32Set authorizedDomains;
+        EnumerableSet.Bytes32Set authorizedRedirectUris;
+        VersionedApp[] versionedApps;
+        address manager;
+        string name;
+        string description;
+    }
+
+    struct AppStorage {
+        mapping(uint256 => App) appIdToApp;
+        mapping(address => EnumerableSet.UintSet) managerAddressToAppIds;
+        mapping(address => uint256) delegateeAddressToAppId;
+        mapping(bytes32 => string) authorizedDomainHashToDomain;
```

Comment:
enumeration is only necessary because of `getAppById`, correct?

Reply:
If this comment is referring to `managerAddressToAppIds`, I think we can replace the EnumerableSet with just `uint256[]`. We never iterate over this mapping value, we just return the entire array for the view facet function `getAppsByManager`